custodial-copy-reconciler/src/main/scala/uk/gov/nationalarchives/reconciler/Database.scala

```diff
@@ -0,0 +1,128 @@
+package uk.gov.nationalarchives.reconciler
+
+import cats.effect.Async
+import cats.implicits.*
+import doobie.free.connection.ConnectionIO
+import doobie.implicits.*
+import doobie.util.Write
+import doobie.util.log.LogHandler
+import doobie.util.transactor.Transactor
+import doobie.util.transactor.Transactor.Aux
+import doobie.{Query0, Update}
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+import uk.gov.nationalarchives.reconciler.Database.TableName.{ActualCosInPS, ExpectedCosInPS}
+import uk.gov.nationalarchives.utils.Utils.given
+
+import java.util.UUID
+
+trait Database[F[_]]:
+  def writeToActuallyInPsTable(cosInPS: List[PreservicaCoRow]): F[Unit]
+  def writeToExpectedInPsTable(expectedCosInPS: List[OcflCoRow]): F[Unit]
+  def checkIfPsCoInCc(coInPS: PreservicaCoRow): F[List[String]]
+  def checkIfCcCoInPs(coInCC: OcflCoRow): F[List[String]]
+
+object Database:
+  enum TableName:
+    case ExpectedCosInPS, ActualCosInPS
+
+  def apply[F[_]: Async](using configuration: Configuration): Database[F] = new Database[F] {
+
+    val xa: Aux[F, Unit] = Transactor.fromDriverManager[F](
+      driver = "org.sqlite.JDBC",
+      url = s"jdbc:sqlite:${configuration.config.databasePath}",
+      logHandler = Option(LogHandler.jdkLogHandler)
+    )
+
+    override def writeToExpectedInPsTable(expectedCosInPS: List[OcflCoRow]): F[Unit] = {
+      val deleteSql = "delete from ExpectedCosInPS where coRef = ?"
```

Comment:
I don't think we need to delete first. There shouldn't be duplicate IDs but if there were, we'd want to know about it. Same for both tables.

Reply:
So then should I make the CO the primary ID?